src/database/config.rs

```diff
@@ -0,0 +1,85 @@
+
+use std::collections::{HashSet, HashMap};
+
+use serde_derive::{Serialize, Deserialize};
+use serde::Serialize;
+use serde::Serializer;
+use serde::Deserialize;
+use serde::Deserializer;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum RankingOrdering {
+    Asc,
+    Dsc
+}
+
+impl Serialize for RankingOrdering {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        serializer.serialize_str(match *self {
+            RankingOrdering::Asc => "asc",
+            RankingOrdering::Dsc => "dsc",
+        })
+    }
+}
+
+impl<'de> Deserialize<'de> for RankingOrdering {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: Deserializer<'de>
+    {
+        let s = String::deserialize(deserializer)?;
+        Ok(match s.as_str() {
+            "asc" => RankingOrdering::Asc,
+            "dsc" => RankingOrdering::Dsc,
+            _ => RankingOrdering::Dsc
```

Comment:
It is not correct to accept any string without any error, if the string does not match "asc" or "dsc" so we need to return at least `D::Error::custom("invalid ordering")`.